components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -1288,11 +1295,27 @@ public ServiceProvider getServiceProviderByClientId(String clientId, String clie
     public ImportResponse importSPApplication(SpFileContent spFileContent, String tenantDomain, String username,
                                               boolean isUpdate) throws IdentityApplicationManagementException {
 
+        return importSPApplication(spFileContent, tenantDomain, username, MEDIA_TYPE_XML, isUpdate);
+    }
+
+    public ImportResponse importSPApplication(SpFileContent spFileContent, String tenantDomain, String username,
+                                              String fileType, boolean isUpdate)
+            throws IdentityApplicationManagementException {
+
         if (log.isDebugEnabled()) {
             log.debug("Importing service provider from file " + spFileContent.getFileName());
         }
 
-        ServiceProvider serviceProvider = unmarshalSP(spFileContent, tenantDomain);
+        ServiceProvider serviceProvider;
+
+        if (fileType.contains("yaml")) {
+            serviceProvider = deserializationSP(spFileContent, tenantDomain);
+        } else if (fileType.contains("json")) {
```

Comment:
As a best practices, use of constant file for the defined constants and use it whenever needed. Check this throughout the PR.

Reply:
I have defined the strings as constants and used them throughout the PR as suggested.